resource.go

```diff
@@ -59,7 +63,21 @@ func LoadResourceFromPath(path string) (Resource, error) {
 
 // LoadResourceFromURLString creates a new StaticResource in memory using the body of the specified URL.
 func LoadResourceFromURLString(urlStr string) (Resource, error) {
-	res, err := http.Get(urlStr)
+	ctx, cancel := context.WithTimeout(context.Background(), httpReqTimeout)
+	defer cancel()
+
+	return LoadResourceFromURLStringWithCtx(ctx, urlStr)
+}
+
+// LoadResourceFromURLStringWithCtx creates a new StaticResource in memory using the body of the specified URL.
+// The provided context will be passed the the http request.
+func LoadResourceFromURLStringWithCtx(ctx context.Context, urlStr string) (Resource, error) {
```

Comment:
I don't think that this new function adds much utility, if someone understand `Context` then they can probably manage the connection as well.
Just FYI the `LoadResourceFromURLString` will be deprecated some time soon as we move it to `fyne.URI` and the `storage` package (once http and https are in the core repository code).

Reply:
> then they can probably manage the connection as well

Don't see how they should do that as there is no access exposed to the http connection.

Reply:
I was not proposing re-using an http connection, just that reading bytes from a URL is relatively straight forward - this helper was for people who don't want the hassle.
I guess when I meant "the connetion" I was meaning conceptually not the specific variable we use.